app/controllers/application_controller.rb

```diff
@@ -1,2 +1,21 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+  include Pundit
+  before_action :authenticate_user!
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+  def user_not_authorized
+    redirect_to(
+      request.referrer || root_path,
+      alert: t('menus_messages.new.messages.not_admin')
+    )
+  end
+
+  def require_login
+    redirect_to new_user_registration_path unless current_user.id
```

Comment:
if you'd meet the situation when your user has no id, then you'd understand in a while that you have no user at all, even to call `.id` :)